src/hooks/useRepository.ts

```diff
@@ -0,0 +1,27 @@
+import { useQuery } from "react-query";
+import axios from "axios";
+import { FavoriteData, RepoResponse, Repository } from "../interfaces";
+
+export default function useRepository(favorite: FavoriteData) {
+  let searchString = "https://api.github.com/repos/" + favorite.fullname;
+
+  const { status, data, error } = useQuery(
```

Comment:
```suggestion
  const { status, data, error } = useQuery<Repository>(
```

Your data will infer type `Repository | undefined` then